patterns/guaranteed_processor.py

```diff
@@ -0,0 +1,163 @@
+        with lock:
+            self._receipt_count += 1
+            print(f"\npublish_receipt:\n {self.receipt_count}\n")
+
+# Broker Config. Note: Could pass other properties Look into
+broker_props = {
+    "solace.messaging.transport.host": os.environ.get('SOLACE_HOST') or "tcp://localhost:55555,tcp://localhost:55554",
+    "solace.messaging.service.vpn-name": os.environ.get('SOLACE_VPN') or "default",
+    "solace.messaging.authentication.scheme.basic.username": os.environ.get('SOLACE_USERNAME') or "default",
+    "solace.messaging.authentication.scheme.basic.password": os.environ.get('SOLACE_PASSWORD') or ""
+    }
+
+# Build A messaging service with a reconnection strategy of 20 retries over an interval of 3 seconds
+# Note: The reconnections strategy could also be configured using the broker properties object
+messaging_service = MessagingService.builder().from_properties(broker_props)\
+                        .with_reconnection_retry_strategy(RetryStrategy.parametrized_retry(20,3))\
+                        .build()
+
+# Blocking connect thread
+messaging_service.connect()
+print(f'\nMessaging Service connected? {messaging_service.is_connected}')
+
+# Event Handling for the messaging service
+service_handler = ServiceEventHandler()
+messaging_service.add_reconnection_listener(service_handler)
+messaging_service.add_reconnection_attempt_listener(service_handler)
+messaging_service.add_service_interruption_listener(service_handler)
+ 
+# Create a persistent message publisher and start it
+publisher: PersistentMessagePublisher = messaging_service.create_persistent_message_publisher_builder().build()
+
+# Blocking Start thread
+publisher.start_async()
+print(f'Persistent Publisher ready? {publisher.is_ready()}')
+
+# set a message delivery listener to the publisher
+receipt_listener = MessageReceiptListener()
+publisher.set_message_publish_receipt_listener(receipt_listener)
+
+# NOTE: This assumes that a persistent queue already exists on the broker with the same name 
+# and a subscription for topic `solace/samples/python/persistent/processor/output``
+queue_name = 'Q/test/input'
+durable_exclusive_queue = Queue.durable_exclusive_queue(queue_name)
+
+# Build a Receiver with the given topics and start it
+persistent_receiver: PersistentMessageReceiver = messaging_service.create_persistent_message_receiver_builder() \
+                                                                        .with_message_auto_acknowledgement() \
+                                                                        .build(durable_exclusive_queue)
+persistent_receiver.start()
+print(f'Persistent Subscriber is running? {persistent_receiver.is_running()}')
+
+print("\nSend a KeyboardInterrupt to stop processor\n")
+try:
+    # Callback for received messages
+    persistent_receiver.receive_async(ProcessorImpl(publisher))
+    try: 
+        while True:
+            time.sleep(1)
+    except KeyboardInterrupt:
```

Comment:
We're not handling the exception if the queue does not exist 
```
# Handle API exception 
except PubSubPlusClientError as exception:
  print(f'\nMake sure queue {queue_name} exists on broker!')
```
See guaranteed subscriber for more details

Reply:
Done!